mindmeld/components/dialogue.py

```diff
@@ -906,6 +908,20 @@ def _end_slot_fill(self, request, responder, async_mode):
             params=request.params,
         )
 
+        if self._invoked:
+            kwargs = {'domain': request.domain, 'intent': request.intent}
+            name = self._entrance_handler.__name__
+            if "previous_rule" in responder.frame:
+                try:
+                    self._app.app_manager.dialogue_manager.add_dialogue_rule(
+                        name, responder.frame["previous_rule"], **kwargs
+                    )
+                except AssertionError:
+                    self._app.app_manager.dialogue_manager.handler_map[name] = responder.frame[
+                        "previous_rule"
+                    ]
+                responder.frame["previous_rule"] = None
```

Comment:
Can you pop it? `responder.pop("previous_rule", None)`

Reply:
You can also do: `del responder.frame[''previous_rule"]` since `pop` will throw an error if the key does not exist.

Reply:
I thought the second argument (None) in pop is default for that case itself?